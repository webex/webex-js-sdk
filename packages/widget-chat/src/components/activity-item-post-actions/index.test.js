/* eslint-disable max-nested-callbacks */
import React from 'react';
import renderer from 'react-test-renderer';

import ActivityItemPostActions from '.';

describe(`ActivityItemPostActions component`, () => {
  const onDelete = jest.fn();
  describe(`show delete param`, () => {
    const component = renderer.create(
      <ActivityItemPostActions id="abc-123" onDelete={onDelete} showDelete />
  );

    it(`renders properly`, () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe(`without show delete param`, () => {
    const component = renderer.create(
      <ActivityItemPostActions id="abc-123" onDelete={onDelete} showDelete={false} />
  );

    it(`renders properly`, () => {
      expect(component).toMatchSnapshot();
    });
  });

});
