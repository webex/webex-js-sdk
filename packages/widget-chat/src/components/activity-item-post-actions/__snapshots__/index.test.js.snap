exports[`ActivityItemPostActions component show delete param renders properly 1`] = `
<div
  className="post-actions">
  <div
    className="post-actions-item">
    <div
      className="icon-button"
      onClick={[Function handleOnDelete]}>
      <div
        className="icon"
        title="Delete this message" />
    </div>
  </div>
</div>
`;

exports[`ActivityItemPostActions component without show delete param renders properly 1`] = `
<div
  className="post-actions" />
`;
